R/utilities.R

```diff
@@ -805,181 +805,414 @@ trim_scale_expression = function(x){
   return(x)
 }
 
+#' @title Process Regions objects.
+#'
+#' @description INTERNAL FUNCTION to harmonize genomic regions specified as character vectors or data frames.
+#'
+#' @details INTERNAL FUNCTION to harmonize genomic regions specified as character vectors or data frames.
+#'
+#' @param regions Character vector of genomic regions. If neither regions nor regions_df is specified, will use GAMBLR aSHM regions
+#' @param regions_df Data frame of genomic regions with column names "chrom", "start", "end", "name"
+#' @param region_padding Amount to pad the start and end coordinates by. [0]
+#' @param skip_regions Character vector of genes to drop from GAMBLR aSHM regions.
+#' @param only_regions Character vector of genes to include from GAMBLR aSHM regions.
+#' @param projection Specify which genome build to use. [grch37]
+#' 
+#' @import GAMBLR.data
+#'
+#' @return Numeric value.
+#'
+#' @noRd
+#'
+#' @examples
+#' trimmed = trim_scale_expression(2)
+
+process_regions <- function(regions = NULL,
+                            regions_df = NULL,
+                            region_padding = 0,
+                            skip_regions = NULL,
+                            only_regions = NULL,
+                            projection = "grch37") {
```

Comment:
What about an optional "with_chr_prefix" argument? The default could be to strip it (if necessary) and return chromosome names without prefix but if this is enabled, it would ensure they are prefixed (add if necessary).